File: clx-post/src/main/java/com/clx/post/entity/Post.java
package com.clx.post.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 帖子实体。
 */
@Data
public class Post {

    /** 帖子ID */
    private Long id;

    /** 标题 */
    private String title;

    /** 内容 */
    private String content;

    /** 摘要 */
    private String summary;

    /** 作者ID */
    private Long authorId;

    /** 作者名称（冗余） */
    private String authorName;

    /** 分类ID */
    private Long categoryId;

    /** 分类名称（冗余） */
    private String categoryName;

    /** 浏览数 */
    private Integer viewCount;

    /** 点赞数 */
    private Integer likeCount;

    /** 评论数 */
    private Integer commentCount;

    /** 是否置顶 */
    private Boolean isTop;

    /** 是否精华 */
    private Boolean isEssence;

    /** 状态:0正常,1禁用,2草稿 */
    private String status;

    /** 是否删除 */
    private Boolean isDeleted;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}

File: clx-admin-web/src/api/post.ts
// 帖子管理 API

export interface PostAdminVO {
  id: number
  title: string
  content: string
  authorId: number
  authorName: string
  categoryId?: number
  categoryName?: string
  status: number // 0=正常, 1=隐藏, 2=删除
  viewCount: number
  likeCount: number
  commentCount: number
  createdAt: string
}

export interface PostPageRequest {
  title?: string
  status?: number
  authorId?: number
  pageNo: number
  pageSize: number
}

export interface PostPageResponse {
  total: number
  list: PostAdminVO[]
}

import { get, post, put, del } from './index'

export const postAdminApi = {
  /** 分页查询帖子 */
  page: (params: PostPageRequest) => post<PostPageResponse>('/admin/post/page', params),

  /** 获取帖子详情 */
  get: (postId: number) => get<PostAdminVO>(`/admin/post/${postId}`),

  /** 更新帖子状态 */
  updateStatus: (postId: number, status: number) =>
    put<void>(`/admin/post/${postId}/status`, { status }),

  /** 删除帖子 */
  remove: (postId: number) => del<void>(`/admin/post/${postId}`)
}
